### PR TITLE
Sequencer health tx check by Transaction

### PR DESCRIPTION
**Pull request by RodrigoAD**

### CODE REVIEW

File: packages/sources/layer2-sequencer-health/README.md [hunk 1]
```diff
@@ -4,14 +4,16 @@ Adapter that checks the Layer 2 Sequencer status
...
-|           | `ARBITRUM_HEALTH_ENDPOINT` |                            Arbitrum Health Endpoint                            |         |                                              |
-|           |  `OPTIMISM_RPC_ENDPOINT`   |                             Optimism RPC Endpoint                              |         |         https://mainnet.optimism.io          |
-|           | `OPTIMISM_HEALTH_ENDPOINT` |                            Optimism Health Endpoint                            |         | https://mainnet-sequencer.optimism.io/health |
+| Required? |            Name            |                                   Description                                   | Options |                 Defaults to                  |
+| :-------: | :------------------------: | :-----------------------------------------------------------------------------: | :-----: | :------------------------------------------: |
+|    ✅     |   `UNFUNDED_PRIVATE_KEY`   |                            Private key with no funds                            |         |                                              |
```
[the remark this answers is not in the document]
Comment (RodrigoAD) [reply packages/sources/layer2-sequencer-health/README.md:9]: Right! I just made it for the tests and forgot here 😅

File: packages/sources/layer2-sequencer-health/src/network.ts [hunk 2]
```diff
@@ -45,3 +45,65 @@ export const requestBlockHeight = async (network: Networks): Promise<number> =>
...
+      ),
+    )
+  try {
+    Logger.info(`Submitting empty transaction for network: ${network}`)
+    const receipt = await Promise.race([
+      _setTxTimeout(timeout),
+      wallet.sendTransaction(networkTx[network]),
+    ])
+    Logger.info(`Transaction receipt received for network: ${network}`)
+    return (await receipt.wait()).confirmations > 0
```
[the remark this answers is not in the document]
Comment (RodrigoAD) [reply packages/sources/layer2-sequencer-health/src/network.ts:100]: You're right, it should never enter here, but who knows. I'll log some more info

File: packages/sources/layer2-sequencer-health/src/network.ts [hunk 3]
```diff
@@ -45,3 +45,65 @@ export const requestBlockHeight = async (network: Networks): Promise<number> =>
...
+  timeout: number,
+): Promise<boolean> => {
+  const rpcEndpoint = RPC_ENDPOINTS[network]
+  const provider = new ethers.providers.JsonRpcProvider(rpcEndpoint)
+  const wallet = new ethers.Wallet(privateKey, provider)
+
+  // These errors come from the Sequencer when submitting an empty transaction
+  const sequencerOnlineErrors = {
+    [Networks.Arbitrum]: 'gas price too low',
+    [Networks.Optimism]: ''
```
[the remark this answers is not in the document]
Comment (RodrigoAD) [reply packages/sources/layer2-sequencer-health/src/network.ts:61]: Right, is not implemented yet, will add the note

File: packages/sources/layer2-sequencer-health/src/adapter.ts [hunk 4]
```diff
@@ -123,10 +124,22 @@ export const execute: ExecuteWithConfig<ExtendedConfig> = async (request, _, con
   for (let i = 0; i < wrappedMethods.length; i++) {
     const method = wrappedMethods[i]
     const isHealthy = await method(network, config.delta, config.deltaBlocks)
-    // TODO: If unhealthy, doublecheck submitting an empty tx. Not receiving any receipt should confirm that's not healthy
-    if (!isHealthy) return _respond(false)
+    if (!isHealthy) {
+      // TODO: Decide if getStatusByTransaction should determine the final state
```
[the remark this answers is not in the document]
Comment (RodrigoAD) [reply packages/sources/layer2-sequencer-health/src/adapter.ts:128]: We need to decide whether to use it or not. Right now will be useful for debugging the Sequencer behaviour and the accuracy of the previous methods

**Review by aalu1418 — DISMISSED** [review]
👍🏼

**Review by aalu1418 — APPROVED** [review]
👍🏼

**Pull request merged**